storage/engine/mvcc.go

```diff
@@ -497,7 +497,12 @@ func MVCCGet(engine Engine, key roachpb.Key, timestamp roachpb.Timestamp, consis
 		return nil, nil, err
 	}
 
-	return mvccGetInternal(iter, key, metaKey, timestamp, consistent, txn, buf)
+	value, intents, err := mvccGetInternal(iter, key, metaKey, timestamp, consistent, txn, buf)
+	if value == &buf.value {
```

Comment:
The old code used a shallow copy of `MVCCValue.Value`, and this is equivalent. As long as `gogoproto` doesn't reuse buffers across a `Reset()` this is safe (I'm pretty sure that's currently true, although the C++ proto implementation had an optimization like this, so there is some precedent for an optimization that would make shallow copies unsafe)

The returned `value` will always be either `&buf.value` or `nil`, right? Maybe it should be changed to a boolean so there is no chance of the code passing along the returned `value` directly without copying it.